### DISCUSSION TITLE
Address changes - bigger issue

### DISCUSSION THREAD

**Discussion by Tech4TriCounty** [code4recovery/12-step-meeting-list#1428] [General]
We are getting into some difficult territory with the address of our meetings. I originally had a problem with a meeting and thought we had fixed it. Today I am writing because the issue has changed a bit. I published the change on our site but for some reason the folks picking up our feed in the meeting guide are changing the address from the feed. Is there a reason this might happen? Is this our problem or theirs. Either way its ultimately the alcoholics problem because the end result is bad data. How would I address this? Please see the attached images that illustrate the problem. 

I have an account with the meeting guide folks where I can flush and or reload our meetings in the meeting guide.  I can't figure this one out and I don't have even knowledge of the system to really know what is going on. 

I also don't really understand the geocoding that google/the TSML are doing. It seems I can't just edit the data in the sql tables and fix things 

<img width="966" alt="Screenshot 2024-05-27 at 1 29 53 PM" src="https://github.com/code4recovery/12-step-meeting-list/assets/132401769/15493819-6bc5-492f-b14b-eb8dd843469f">

<img width="1137" alt="Screenshot 2024-05-27 at 1 28 59 PM" src="https://github.com/code4recovery/12-step-meeting-list/assets/132401769/7aa59445-f6a1-493f-9611-1db93317ffee">
![IMG_6757](https://github.com/code4recovery/12-step-meeting-list/assets/132401769/64bc9277-6725-4bf4-8425-01be51c66f0d)

**Comment by Tech4TriCounty** [comment]
![IMG_6757](https://github.com/code4recovery/12-step-meeting-list/assets/132401769/16afea5e-c221-4b37-a483-5ad98c90ebfa)

**Comment by ginslo** [comment]
Good afternoon, @Tech4TriCounty. It looks like this location is listed on your TSML list as well as on AA St. Louis with different addresses. See below for the differences in the listings:

```
AA St Louis
https://aastl.org/meetings/group-616-2/
204 W Pitman St
O'Fallon, MO 63366
```

```
Tri-County Intergroup
https://tricountyaa.org/meetings/group-616/
207 E Pitman St
O'Fallon, MO 63366
```

Best suggestion at this point is to collaborate with AA St. Louis to be sure you are both listing matching addresses. It takes some time for changes to propagate to the Meeting Guide, but after some time these differences should resolve. 

Please let me know if this addresses the reported issue.

**Comment by Tech4TriCounty** [comment]
Yes we are aware of that conflict. We coordinate when we find these. Usually when a location moves. However this does not account for the wrong address at the new location 207 E Pitman. The old address 204 W Pitman is the one aastl.org has. When Tom is in the office tomorrow he’ll remove the duplicates he has. Are you saying meeting guide will stop truncating the address we have once the duplicates are removed?  We’ve had it in the past where the two locations were listed until he removed the duplicates but never where the address was not correct. St. Louis now takes their feed from us so as to have the most current info for meetings in our area. That said the hand entered ones have to be discovered. Yours in ServiceBob MTechnology ChairTri-County Intergroup ***@***.***: 636-970-0013C: 314-315-3364    ---- On Mon, 27 May 2024 17:02:04 -0500  ***@***.******@***.***> wrote ----  
Good afternoon, @Tech4TriCounty. It looks like this location is listed on your TSML list as well as on AA St. Louis with different addresses. See below for the differences in the listings: 
AA St Louis 
https://aastl.org/meetings/group-616-2/ 
204 W Pitman St 
O'Fallon, MO 63366 
 
Tri-County Intergroup 
https://tricountyaa.org/meetings/group-616/ 
207 E Pitman St 
O'Fallon, MO 63366 
 
Best suggestion at this point is to collaborate with AA St. Louis to be sure you are both listing matching addresses. It takes some time for changes to propagate to the Meeting Guide, but after some time these differences should resolve. 
Please let me know if this addresses the reported issue.

**Reply by ginslo** [reply]
> Are you saying meeting guide will stop truncating the address we have once the duplicates are removed?  We’ve had it in the past where the two locations were listed until he removed the duplicates but never where the address was not correct.

TBH, I don't know for sure but it should. I suggest seeing how it goes after the two listings (yours and aastl.org) have matching addresses. If the problem persists, we'll take it from there. Note that for issues directly related to the Meeting Guide app, the best path forward is to send a report to appsupport@aa.org as Code For Recovery does not manage the app (you mentioned that you have an account already).

**Comment by Tech4TriCounty** [comment]
Thank you I appreciate the thoughts. This club deals with a lot of newcomers and they were hoping for a quick answer. Holiday weekends are slow to change. ;)  ---- On Mon, 27 May 2024 17:24:22 -0500  ***@***.******@***.***> wrote ----  
 
Are you saying meeting guide will stop truncating the address we have once the duplicates are removed?  We’ve had it in the past where the two locations were listed until he removed the duplicates but never where the address was not correct. 
 
TBH, I don't know for sure but it should. I suggest seeing how it goes after the two listings (yours and aastl.org) have matching addresses. If the problem persists, we'll take it from there. Note that for issues directly related to the Meeting Guide app, the best path forward is to send a report to ***@***.*** as Code For Recovery does not manage the app (you mentioned that you have an account already).

**Comment by ginslo** [comment]
@Tech4TriCounty, quick update after discussion with the C4R team. The first step of matching the two feeds should obviously be done. However that will likely not resolve the issue of the truncated addresses on the Meeting Guide. Solving that will require working directly with the [Meeting Guide](https://meetingguide.helpdocs.io/) folks. So for expediency purposes, I suggest that you open a ticket with them. Let them know of the history and specifically about truncated address issue.

**Comment by Tech4TriCounty** [comment]
OK we cleared the issue of duplicates and got another clue to the problem. So any meeting that has one of the address anomolies gets their address truncated in the meeting guide. 

Our Meetings page: https://tricountyaa.org/meetings/
Our Feed: https://tricountyaa.org/wp-admin/admin-ajax.php?action=meetings

From the feed, the address appears to be ok, but is displayed truncated in the meeting guide and incorrectly with extraneous characters in the TSML admin. It is only addresses like this that are problematic. 
{"id":4827,"name":"212 Club","slug":"212-club-wednesdays-at-1200-pm","updated":"2023-02-09 14:04:45","location_id":12904,"url":"https:\/\/tricountyaa.org\/meetings\/212-club-wednesdays-at-1200-pm\/","day":3,"time":"12:00","time_formatted":"Noon","edit_url":"https:\/\/tricountyaa.org\/wp-admin\/post.php?post=4827&action=edit","types":["O"],"author":"tricountyaa_admin","location":"212 Club","location_url":"https:\/\/tricountyaa.org\/locations\/212-club\/","formatted_address":"207 E Pitman St, O'Fallon, MO 63366, USA","approximate":"no","latitude":38.8069804,"longitude":-90.6976733,"region_id":358,"region":"O'Fallon","regions":["O'Fallon"],"attendance_option":"in_person"}

{"id":11820,"name":"Group 1023","slug":"group-1023-2","updated":"2024-05-28 18:58:36","location_id":11819,"url":"https:\/\/tricountyaa.org\/meetings\/group-1023-2\/","day":2,"time":"18:30","end_time":"19:30","time_formatted":"6:30 pm","edit_url":"https:\/\/tricountyaa.org\/wp-admin\/post.php?post=11820&action=edit","types":["B","C"],"author":"tricountyaa_admin","location":"O'Fallon Christian Church","location_url":"https:\/\/tricountyaa.org\/locations\/ofallon-christian-church-3\/","formatted_address":"9380 Veterans Memorial Pkwy o, O'Fallon, MO 63366, USA","approximate":"no","latitude":38.8017233,"longitude":-90.724995,"region_id":358,"region":"O'Fallon","regions":["O'Fallon"],"group_id":11821,"group":"1023","district":"District 19","district_id":357,"attendance_option":"in_person"},

See attached images.
<img width="925" alt="Screenshot 2024-05-29 at 5 42 55 PM" src="https://github.com/code4recovery/12-step-meeting-list/assets/132401769/ad8d51d4-8a33-491c-a1f3-bc5679c97ad8">
<img width="939" alt="Screenshot 2024-05-29 at 5 41 52 PM" src="https://github.com/code4recovery/12-step-meeting-list/assets/132401769/2214efbf-1ec7-405b-8639-eae76504c9b5">
<img width="938" alt="Screenshot 2024-05-29 at 5 41 06 PM" src="https://github.com/code4recovery/12-step-meeting-list/assets/132401769/cd7fabfa-3eed-4113-b1d3-a35cda37a86b">
<img width="929" alt="Screenshot 2024-05-29 at 5 42 37 PM" src="https://github.com/code4recovery/12-step-meeting-list/assets/132401769/93e62f6c-ab3f-4f33-ae99-b035bd3029f6">

**Comment by Tech4TriCounty** [comment]
These meetings in the meeting guide have their addresses truncated. 
![IMG_0021](https://github.com/code4recovery/12-step-meeting-list/assets/132401769/a3b46e9c-42d4-478d-96f9-0a1ec7ebc6c7)
![IMG_0020](https://github.com/code4recovery/12-step-meeting-list/assets/132401769/771a22d3-d82c-4102-850d-6786949856f7)

**Reply by Tech4TriCounty** [reply]
Meetings with a unhinkey address display properly in the meeting guide. One other hint to the problem is that if you click the directions in the meeting guide it takes you to the correct address. I don't know all the where and whats of the geocoding and don't mind taking it to the folks at AAWS but I don't think its just their issue. I think they could compensate but the issue seems to lay with how TSML/Google do the address work.

**Reply by ginslo** [reply]
Thanks for the info @Tech4TriCounty. Good progress. To complete the loop, it would be helpful if you could take it to GSO who can add the street number to the address for you. Note that C4R is still looking into this to rule out any coding issues around this address string.

**Reply by SIA-WEBSUP** [reply]
@Tech4TriCounty, something you might try entirely on your end is:

1.  signing in to your 'meeting guide entity dashboard' ( https://meetingguide.org/login )
2.  flush the cache
3.  reload your feed

Please let us know if this fixes the problem.
If it doesn't, as Gary already suggested, please consider contacting the meeting guide support team directly to see if they can help resolve this issue ( appsupport@aa.org  )

![image](https://github.com/code4recovery/12-step-meeting-list/assets/63924390/ae544114-8e98-4a23-a359-bb3e3890df9a)

**Reply by ginslo** [reply]
Good afternoon @Tech4TriCounty , when you have a moment, please try editing those O'Fallon meetings again to see if the address works better now. We tweaked some things in the latest version of the plugin. Then let us know if you see the address  appear correctly on the app. Thanks.

**Reply by Tech4TriCounty** [reply]
I did this and the address appears to be correct in our systems accross the board however the issue with the meeting guide is still present. I will contact them concerning this issue and see what input they may have.

**Reply by SIA-WEBSUP** [reply]
I wanted to double check... did you flush the meeting guide app's geocoding cache for your site via the 'meeting guide entity dashboard'?
    OR   did you 'edit' the meetings on your website via the TSML plugin's admin screen?
    OR   both ?

**Comment by Tech4TriCounty** [comment]
BothYours in ServiceBob MTechnology ChairTri-County Intergroup ***@***.***: 636-970-0013C: 314-220-4350    ---- On Mon, 03 Jun 2024 13:59:38 -0500  ***@***.******@***.***> wrote ----  
I wanted to double check... did you flush the meeting guide app's geocoding cache for your site via the 'meeting guide entity dashboard'? 
OR   did you 'edit' the meetings on your website via the TSML plugin's admin screen? 
OR   both ?

**Comment by Tech4TriCounty** [comment]
Looks like we are all good now. The TSML addressing and the meeting guide addressing are fixed. Meeting Guide app support was able to locate the issue with meetings and correct it.

Here is what they reported:
"This issue was actually related to a code change for the App that had an unexpected consequence of dropping the street numbers when an address was re-geocoded (so if a meeting was updated for instance).

We only became aware of it in the last few days and so our development team made a fix to the code.  This issue had nothing to do with TSML plugin."

**Reply by ginslo** [reply]
Hey @Tech4TriCounty that's great news. Really appreciate the feedback and for your patience as we worked through this somewhat multi-layered issue.  Thank you for your service as well!
